File: performance-comparison/benchmarks/io/python/file_write.py
import sys, os

n    = int(sys.argv[1]) if len(sys.argv) > 1 else 1048576
path = "/tmp/bench_io_write_py.bin"
buf  = b'\xAB' * 4096
with open(path, 'wb') as f:
    written = 0
    while written < n:
        chunk = min(4096, n - written)
        f.write(buf[:chunk])
        written += chunk
os.remove(path)
print(n)

File: performance-comparison/benchmarks/io/clojure/file_write.clj
(let [n   (Long/parseLong (first *command-line-args*))
     path "/tmp/bench_io_write_clj.bin"
     buf  (byte-array 4096 (byte 0xAB))]
  (with-open [f (java.io.FileOutputStream. path)]
    (loop [written 0]
      (if (>= written n)
        (println written)
        (let [chunk (min 4096 (- n written))]
          (.write f buf 0 (int chunk))
          (recur (+ written chunk))))))
  (java.io.File. path) (.delete (java.io.File. path)))
